ghttp/ghttpProcess.c

```diff
@@ -1257,10 +1257,10 @@ static int ghiParseChunkSize
 	GSI_UNUSED(len);
 
 	rcode = sscanf(header, "%x", &num);
-	if(rcode != 1)
+	if(rcode != 1 || num > INT_MAX)
 		return -1;
 
-	return num;
+	return (int)num;
```

Comment:
According to https://en.wikipedia.org/wiki/Chunked_transfer_encoding, the chunk data must be >-1, I would change the function ghiParseChunkSize to be unsigned int and remove this cast

Reply:
Problem here is `-1` value used as error indicator.

If we return `UINT_MAX` as error, it requires to change caller's `connection->chunkBytesLeft` type, which is `int`.
Which in turn requires to change internal parameters to `unsigned int` and `len` in `ghiProcessIncomingFileData`, and so on.

Reply:
hmm true I forgot to see the use of that (whicle perhaps 0 might be an error as well 🤔 ), better to keep it this way then